### PR TITLE
DVC-7156 add VariableValue method

### PR DESCRIPTION
**Pull request by jonathannorris**

### CODE REVIEW

File: client_test.go [hunk 1]
```diff
@@ -266,6 +274,11 @@ func TestClient_VariableLocalProtobuf_UserWithCustomData(t *testing.T) {
 		t.Fatal("Expected variable to be equal to expected variable")
 	}
 	fmt.Println(variable)
+
+	variableValue, err := c.VariableValue(user, "test", true)
```
Comment (JamieSinn) [inline client_test.go:278]: You're not checking the value of `err` here

File: client_test.go [hunk 2]
```diff
@@ -218,6 +222,11 @@ func TestClient_VariableLocalProtobuf(t *testing.T) {
...
```
Comment (JamieSinn) [inline client_test.go:226]: You're not checking the value of `err` here

File: client.go [hunk 3]
```diff
@@ -276,6 +276,15 @@ func (c *Client) AllFeatures(user User) (map[string]Feature, error) {
 	return nil, c.handleError(r, rBody)
 }
 
+func (c *Client) VariableValue(userdata User, key string, defaultValue interface{}) (result interface{}, err error) {
+	variable, err := c.Variable(userdata, key, defaultValue)
+	if err != nil {
+		return Variable{}, err
```
Comment (JamieSinn) [inline client.go:282]: Returning an empty variable object isn't a good idea here - as it'll default to an empty object with no default value set.

**Review by JamieSinn — CHANGES_REQUESTED** [review]

File: client.go (hunk 3 above)
Comment (jonathannorris) [reply client.go:282]: that's what we do in the `Variable` method below that I copied from.

Comment (jonathannorris) [reply client.go:282]: I guess I should just return whatever `Variable()` returns

File: client.go [hunk 4]
```diff
@@ -279,7 +279,7 @@ func (c *Client) AllFeatures(user User) (map[string]Feature, error) {
 func (c *Client) VariableValue(userdata User, key string, defaultValue interface{}) (result interface{}, err error) {
 	variable, err := c.Variable(userdata, key, defaultValue)
 	if err != nil {
-		return Variable{}, err
+		return variable.Value, err
 	}
 
 	return variable.Value, nil
```
Comment (JamieSinn) [inline client.go]: This can just be `return variable.Value, err` as we don't care about the actual values of `err`

File: client.go [hunk 5]
```diff
@@ -276,13 +276,35 @@ func (c *Client) AllFeatures(user User) (map[string]Feature, error) {
...
+
+    -@return interface{}
+*/
+func (c *Client) VariableValue(userdata User, key string, defaultValue interface{}) (result interface{}, err error) {
+	variable, err := c.Variable(userdata, key, defaultValue)
+	if err != nil {
+		return variable.Value, err
+	}
+
+	return variable.Value, nil
```
Comment (JamieSinn) [inline client.go:296]:
```suggestion
	return variable.Value, err
```

**Review by chris-hoefgen — APPROVED** [review]

**Review by JamieSinn — APPROVED** [review]

**Pull request merged**